best-practice/definitions.ts

```diff
@@ -0,0 +1,29 @@
+import { IAllBestPractice, IBestPractice } from "./types.ts";
+
+const github: IBestPractice[] = [
+  {
+    name: "Default Branch Name",
+    owner: "[@guardian](https://github.com/orgs/guardian/teams/all)",
+    description:
+      "The default branch name should be `main`.<br>See the [master-to-main tool](https://github.com/guardian/master-to-main/blob/main/migrating.md).",
+    howToCheck: "Manual. View the repository on https://github.com",
+    howToExempt: "N/A",
+  },
+];
```

Comment:
I would consider using a constant assertion to narrow this object's type to an exact version:

```ts

const github = [
/* the object */
] as const satisfies IBestPractice;
```

Reply:
`as const satisfies IBestPractice[];` fails to compile as the `as const` makes everything readonly:

```log
error: TS1360 [ERROR]: Type 'readonly [{ readonly name: "Default Branch Name"; readonly owner: "[@guardian](https://github.com/orgs/guardian/teams/all)"; readonly description: "The default branch name should be `main`.<br>See the [master-to-main tool](https://github.com/guardian/master-to-main/blob/main/migrating.md)."; readonly howToCheck: "Ma...' does not satisfy the expected type 'IBestPractice[]'.
  The type 'readonly [{ readonly name: "Default Branch Name"; readonly owner: "[@guardian](https://github.com/orgs/guardian/teams/all)"; readonly description: "The default branch name should be `main`.<br>See the [master-to-main tool](https://github.com/guardian/master-to-main/blob/main/migrating.md)."; readonly howToCheck: "Ma...' is 'readonly' and cannot be assigned to the mutable type 'IBestPractice[]'.
] as const satisfies IBestPractice[];
```

I'm not sure I understand the advantage of using `satisfies` here. Could you elaborate please?

Reply:
Using `satisfies` ensures that the input data is both narrowed but also adheres to the expected shape. Without it, a future contributor might add an unexpected key in on of the arrays' objects, and `as const` would not complain.

For data that is truly immutable I think it's the best of both declaring interfaces and getting an exact type.

In this case we would probably need to make the interface `readonly`, which is more correct anyways as we do not want any method to inadvertently mutate it.